docs/design/environment/enviroment_management.md

```diff
@@ -0,0 +1,153 @@
+# #   default: # default configurations, applied to all projects
+# #     <ModuleInput_Field1>: <value1>
+# #     <ModuleInput_Field2>: <value2>
+# #     ...
+# #   <patcherName>: #patcher configurations, applied to the projects assigned in projectSelector
+# #     <ModuleInput_Field1>: <value1_Override>
+# #     ...
+# #     projectSelector:
+# #     - <projectName1>
+# #     ...
+module:
+  database:
+    default:
+      provider: aws
+      size: 20
+      instanceClass: db.t3.micro
+      securityIPs:
+      - 10.0.0.0/18
+    smallClass:
+      size: 50
+      instanceClass: db.t3.small
+      projectSelector:
+      - foo
+      - bar
+    largeClass:
+      instanceClass: db.t3.large
+      projectSelector:
+      - foo
+      - bar
+```
+
+The environment configuration file only includes module configs for now, whose key is `module`, and value is a yaml dictionary containing specified module configs whose key is the module name.
+
+A module config is composed of a default configs and several patchers, where the name of patcher must not be default. For default config or a certain patcher, the structure is also a yaml dictionary, and the key must be the module input field name.
+
+Following the writing format standard above, we can achieve:
+
+- Clearly define the modules' inputs in an environment;
+- Support customized configurations for some projects;
+- Use a unify function to combine schema configurations and environment configurations to form the entire inputs for all modules. 
+
+### Commands
+
+Kusion provides the management of environment by sub command `kusion env`, the designed command is shown as below.
+
+#### kusion env create \<name> -f \<configFilePath>
+
+Create an environment.
+
+#### kusion env update \<name> -f \<configFilePath>
+
+Create an environment configuration.
+
+#### kusion env delete \<name>
+
+Delete an environment configuration.
+
+#### kusion env list 
+
+List all the initialized environments.
+
+#### kusion env show \<name>
+
+Show the configurations of specified environment.
+
+There are the user cases:
+
+- **Individual developer**: The user creates environments as needed (e.g. dev, pre and prod), and writes AppConfiguration codes in corresponding project and stacks locally. When applying a specified stack, set kubeconfig and provider authentication info, and corresponding environment configurations are used automatically to build intent.
+- **Collaboration Scenario**: An application developer gets the names of environments in use, then write project and stack AppConfiguration codes locally, with stack name the same as environment name. In CI/CD pipeline, create environments according to the changed stacks, set kubeconfig and provider authentication info, then apply a specified stack. In this scenario, the full environment configurations should be stored by platform team for now. After the support of centralized storage backend, the storage work of environment configuration can be handled by Kusion.
```

Comment:
> In CI/CD pipeline, create environments according to the changed stacks, set kubeconfig and provider authentication info, then apply a specified stack.

Just thinking, if the pipeline code needs to pull down the file manually and perform `kusion env create` every time the pipeline runs, conceptually it kind of loses the point of an "environment" - which is an abstraction layer for the configs/secrets that are associated with it (Instead, the pipeline code directly manages the configs and secrets that are needed for Kusion modules).

Reply:
That's because we don't provide centralized storage for now. But for the command, we give a systematized design, and provide a good and simple environment management experience.